### PR TITLE
Connect to Cloud instances with computed hostname

### PR DESCRIPTION
**Pull request by fantix**
This will currently break the cloud nightly test because:
1. certificate issues.
2. `api.g.<dns-zone>` is not resolvable.

### CODE REVIEW

File: src/cloud/client.rs [hunk 1]
```diff
@@ -5,12 +5,14 @@ use std::io;
...
 use std::time::Duration;
 
+use anyhow::Context;
 use surf::http::auth::{AuthenticationScheme, Authorization};
 
 use crate::options::CloudOptions;
 use crate::platform::config_dir;
 
 const EDGEDB_CLOUD_BASE_URL: &str = "https://free-tier0.ovh-us-west-2.edgedb.cloud";
+const EDGEDB_CLOUD_DEFAULT_DNS_ZONE: &str = "ovh-us-west-2.edgedb.cloud";
```
Comment (elprans) [inline src/cloud/client.rs:15]: These are obsolete.  The default API URL should be `https://api.g.<dns-zone>` and there is no `dns-zone` default, it's always determined from the token.

Comment (fantix) [reply src/cloud/client.rs:15]: So the user should provide a `dns-zone` in `edgedb cloud login`?

Comment (elprans) [reply src/cloud/client.rs:15]: Ah, good point.  Let's use `aws.edgedb.cloud` as the default for logins for now.

File: src/cloud/client.rs [hunk 2]
```diff
@@ -5,12 +5,13 @@ use std::io;
...
 use std::time::Duration;
 
+use anyhow::Context;
 use surf::http::auth::{AuthenticationScheme, Authorization};
 
 use crate::options::CloudOptions;
 use crate::platform::config_dir;
 
-const EDGEDB_CLOUD_BASE_URL: &str = "https://free-tier0.ovh-us-west-2.edgedb.cloud";
+const EDGEDB_CLOUD_DEFAULT_DNS_ZONE: &str = "aws.edgedb.cloud";
```
Comment (elprans) [inline src/cloud/client.rs:14]: We still want the ability to override the complete API host address with `EDGEDB_CLOUD_BASE_URL`.

**Review by elprans — APPROVED** [review]

**Pull request merged**